nvtx.hpp

```diff
@@ -0,0 +1,1810 @@
+/*
+ *  Copyright 2019 Jacob Hemstad
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+#pragma once
+
+#include <nvToolsExt.h>
+
+#include <string>
+
+/**
+ * @file nvtx.hpp
+ *
+ * @brief Provides C++ constructs making the NVTX library safer and easier to
+ * use with zero overhead.
+ */
+
+/**
+ * \mainpage
+ * \tableofcontents
+ * \section Overview
+ *
+ * The NVTX library provides a set of functions for users to annotate their code
```

Comment:
I would add a quick start section here (or at least "above the fold") that has a minimal  example that will work for many use cases. This library has a lot of documentation for something that users will want to spend very little time learning and using.